File: packages/dart/barnard/ios/barnard/Sources/barnard/BarnardV2Policy.swift
// Copyright 2024-2026 The Greeting Inc. All rights reserved.
// Use of this source code is governed by a BSD-style license.

#if canImport(BarnardCore)
import BarnardCore
#endif

enum BarnardV2Policy {
  static func shouldServeGattDisplayId(eventCode: String?) -> Bool {
    BarnardCoreV2Policy.shouldServeGattDisplayId(eventCode: eventCode)
  }

  struct KnownPeerWindow {
    let enin: UInt32

    func matches(_ currentEnin: UInt32) -> Bool {
      BarnardCoreV2Policy.shouldEmitRssiUpdate(
        cachedPeerEnin: enin,
        currentEnin: currentEnin
      )
    }
  }

  static func shouldEmitRssiUpdate(
    cachedPeerEnin: UInt32,
    currentEnin: UInt32
  ) -> Bool {
    BarnardCoreV2Policy.shouldEmitRssiUpdate(
      cachedPeerEnin: cachedPeerEnin,
      currentEnin: currentEnin
    )
  }
}

File: packages/react-native/barnard/android/src/main/kotlin/network/greeting/barnard/BarnardV2Policy.kt
package network.greeting.barnard

internal object BarnardV2Policy {
    fun shouldServeGattDisplayId(eventCode: String?): Boolean {
        return !eventCode.isNullOrEmpty()
    }

    data class KnownPeerWindow(val enin: Long) {
        fun matches(currentEnin: Long): Boolean = enin == currentEnin
    }

    fun shouldEmitRssiUpdate(cachedPeerEnin: Long, currentEnin: Long): Boolean {
        return KnownPeerWindow(cachedPeerEnin).matches(currentEnin)
    }
}
